packages/kbn-securitysolution-utils/src/path_validations/index.ts

```diff
@@ -37,11 +37,15 @@ export type TrustedAppEntryTypes = 'match' | 'wildcard';
  * such as:
  * c:\home\lib\dmp.dmp
  * c:\home\lib\my-binary-app-+/ some/  x/ dmp.dmp
+ * c:\home\lib\my-binary-app-+/ some/  x/ dmp.and.dmp
+ * * c:\home\lib\my-binary-app-+/ some/  x/ dmp.and-a10.dmp
  * /home/lib/dmp.dmp
  * /home/lib/my-binary-app+-\ some\  x\ dmp.dmp
+ * /home/lib/my-binary-app+-\ some\  x\ dmp.and.dmp
+ * /home/lib/my-binary-app+-\ some\  x\ dmp.and-a10.dmp
  */
-export const WIN_EXEC_PATH = /(\\[-\w]+|\\[-\w]+[\.]+[\w]+)$/i;
-export const UNIX_EXEC_PATH = /(\/[-\w]+|\/[-\w]+[\.]+[\w]+)$/i;
+export const WIN_EXEC_PATH = /(\\[-\.\w ]+)$/i;
```

Comment:
I'm not expert in regex, but does `\w` match Unicode and special ASCII codes above 128?
I've found the explanation `\w matches any word character (equivalent to [a-zA-Z0-9_])`
Windows file name can contain a lot of more characters, but also it cannot contain certain things like using two consecutive dots `..`
https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file

I have no opinion of how fool-proof it should be, as this is only to extract the file name part from the given path. I'd even say that it should suffice to extract just the string after the last `\`, provided it's non-empty.

Reply:
This regex is actually not to extract the filename but to check if string after the last `\` (in windows) or `/` (in a unix system) has no wildcards. Currently, we use node's built-in `path` to extract the filenames if the filename is without a wildcard.

I see your point about the Unicode chars. I'll see what I can whip up to cover those cases.

Reply:
what about doing what you mean `/(\\[^*?]+)$/i`

Reply:
Thanks for looking into this @intxgo . 

`/(\\[^*?]+)$/i` works but unfortunately is not [ReDos safe](https://devina.io/redos-checker). 

I came up with the following which is safer.

`/(\\[-\.\w \p{L}]+)$/iu`

**Question:**
Should we cover this kind of naming also for UNIX systems?

Reply:
@intxgo done 7e1a634013a657d9a80e4bd613dea6a71e62bd77

Reply:
Don't forget about other valid characters
```
c:\anything\*\file.exe-2.bck
c:\anything\*\file.exe-?2.bck
c:\anything\*\file.exe*2.bck
c:\anything\*\file.exe-2.bck您正在访问的域名可以转
c:\anything\*\test.txt
c:\anything\*\test!.txt
c:\anything\*\test!#.txt
c:\anything\*\test~1.txt
c:\anything\*\test$.txt
```
... etc

Sure, Linux also allows many special ASCII characters inside paths, as well as Unicode.

Reply:
on a second though, why not extracting the match with `/\\(.+)$/i` and then just do a simple linear find for '*' and '?'. That's get you covered and is ReDos safe.

Reply:
@ashokaditya Limiting the string length would mitigate ReDos. There could be many FS on Linux, but it seems most of them keep a sensible file name length limit ~260 characters. I'll leave this research/decision to you. I think it's far less probable to deal with an FS allowing very long file names than to have file name with non `\w` characters in it, thus if you take out last ~300 characters from the path to check with regex it should be fine.

Reply:
8161223cd22622c50111f121ca46a12290ef0d83